testing/integration/controller-wifi.nix

```diff
@@ -0,0 +1,331 @@
+
+            dhcp-option = [
+                "3,10.0.9.2" # gateway, does not exist
+                "6,10.0.9.3" # DNS, does not exist
+            ];
+            dhcp-range = "10.0.9.30,10.0.9.99,1h";
+        };
+
+        # disable the local DNS caching proxy in connman to avoid conflicts with
+        # dnsmasq
+        services.connman.extraFlags = [ "--nodnsproxy" ];
+
+
+        # === Simulated wireless access points
+
+        # enable 802.11 simulation
+        boot.kernelModules = [ "mac80211_hwsim" ];
+
+        # wireless access points
+        services.hostapd = {
+          enable = true;
+          # note: do not change this to wlan1 or other id, weird failures appear
+          radios.wlan0 = {
+            band = "2g";
+            countryCode = "US";
+            networks = {
+              wlan0 = {
+                ssid = "test-ap-sae";
+                bssid = "02:00:00:00:00:00";
+                authentication = {
+                  mode = "wpa3-sae";
+                  saePasswords = [ { password = "reproducibility"; } ];
+                };
+              };
+              wlan0-1 = {
+                ssid = "test-ap-mixed";
+                bssid = "02:00:00:00:00:01";
+                authentication = {
+                  mode = "wpa3-sae-transition";
+                  saeAddToMacAllow = true;
+                  saePasswordsFile = pkgs.writeText "password" "reproducibility";
+                  wpaPasswordFile = pkgs.writeText "password" "reproducibility";
+                };
+              };
+
+              # connman hangs when connecting to this AP, nothing in the logs
+              # neither from connman nor from wpa_supplicant ¯\_(ツ)_/¯
+              #wlan0-2 = {
+              #  ssid = "test-ap-wpa2";
+              #  authentication = {
+              #    mode = "wpa2-sha256";
+              #    wpaPassword = "reproducibility";
+              #  };
+              #  bssid = "02:00:00:00:00:02";
+              #};
+
+              wlan0-3 = {
+                ssid = "test-ap-open";
+                bssid = "02:00:00:00:00:03";
+                authentication = {
+                  mode = "none";
+                };
+              };
+
+              # connecting to this AP fails with `invalid-key`
+              # (even though the reason is diferent). This is
+              # because wpa_supplicant responds with CTRL-EVENT-AUTH-REJECT
+              # status_code=1, which is "unspecified reason" and so
+              # connman does a buest guess.
+              #wlan0-4 = {
+              #  ssid = "bad-ap-blocked";
+              #  bssid = "02:00:00:00:00:04";
+              #  authentication = {
+              #    mode = "wpa3-sae";
+              #    saePasswords = [ { password = "reproducibility"; } ];
+              #  };
+              #  macAcl = "allow"; # allows only listed clients
+              #  macAllow = [ ] ; # deny everything
+              #};
```

Comment:
This seems like good info to capture, but the commented out code makes its status unclear to me.

Could this even be a test that runs, but we explain the unexpected message in the test code?

Reply:
It's quite an esoteric case, not really worthy of a test - I have simply removed it in 4b38813